src/js/utils/Select.jsx

```diff
@@ -129,6 +130,30 @@ class Select extends Component {
     }
   }
 
+  sortOptionsByChecked(options, checkedValues) {
+    const groupedByChecked =
+      options?.length ?
```

Comment:
as you probably know, I'm a fan of functional programming and immutable approach, so that's why I prefer the `reduce` approach instead of `forEach` function which is a void function.
If it doesn't bother you that much, I'd love to stay with the `reduce` version, as for `if`, I can agree and will change it :)